This file is part of MXE. See LICENSE.md for licensing information.

Contains ad hoc patches for cross building.

From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Gregorio Litenstein <g.litenstein@gmail.com>
Date: Thu, 31 Aug 2017 18:35:23 -0300
Subject: [PATCH] Update patch-3 for opencv-3.3.0

Originally From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Boris Nagaev <bnagaev@gmail.com>
Date: Sun, 20 Dec 2015 18:14:56 +0300
Subject: [PATCH] fix CMake error if a directory contains ++

See https://github.com/mxe/mxe/issues/1091

diff --git a/cmake/OpenCVModule.cmake b/cmake/OpenCVModule.cmake
index 1111111..2222222 100644
--- a/cmake/OpenCVModule.cmake
+++ b/cmake/OpenCVModule.cmake
@@ -491,12 +491,16 @@ function(__ocv_resolve_dependencies)
   # process private deps
   foreach(m ${OPENCV_MODULES_BUILD})
     foreach(d ${OPENCV_MODULE_${m}_PRIVATE_REQ_DEPS})
-      if(NOT (";${deps_${m}};" MATCHES ";${d};"))
+      # can't use MATCH because file paths can include regexp special chars
+      string(FIND ";${deps_${m}};" ";${d};" d_offset)
+      if(${d_offset} EQUAL -1)
         list(APPEND deps_${m} ${d})
       endif()
     endforeach()
     foreach(d ${OPENCV_MODULE_${m}_PRIVATE_OPT_DEPS})
-      if(NOT (";${deps_${m}};" MATCHES ";${d};"))
+      # can't use MATCH because file paths can include regexp special chars
+      string(FIND ";${deps_${m}};" ";${d};" d_offset)
+      if(${d_offset} EQUAL -1)
         if(HAVE_${d} OR TARGET ${d})
           list(APPEND deps_${m} ${d})
         endif()
